blog/views.py

```diff
@@ -45,4 +59,45 @@ def post(self, request, format=None):
             auth.logout(request)
             return Response({ 'success': 'Loggout Out' }, status=200)
         except:
-            return Response({ 'error': 'Something went wrong when logging out' }, status=401)
\ No newline at end of file
+            return Response({ 'error': 'Something went wrong when logging out' }, status=401)
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class SignUpView(APIView):
+    permission_classes = ()
+    serializer_class = SignUpSerializerForm
+    
+    def post(self, request, format=None):   
+        try:
+            data = json.loads(self.request.data)
+        except:
+            data = self.request.data
+
+        username = data["username"]
+        password =  data["password"]
+        displayName = data["displayName"]
+        profileImage = data["profileImage"]
+        github = data["github"]
+        
+
+        if not valid_url(github):
+            return Response({"error": "Error Occured"}, status=400)
```

Comment:
Since these are called separately, could we have a slightly more descriptive error message?